### PR TITLE
Disabled profiling on connection opening when profiling is disabled

### PR DESCRIPTION
**Pull request by njasm**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | no
| Fixed issues  | #14877

### CODE REVIEW

**Review by samdark — APPROVED** [review]

**Comment by samdark** [comment]
@yiisoft/reviewers need reviews.

**Review by rob006 — APPROVED** [review]
Docs of `enableProfiling` should be updated. See https://github.com/yiisoft/yii2/issues/14877#issuecomment-336674029

**Comment by samdark** [comment]
Yes. @njasm would you please handle docs as well?

**Comment by njasm** [comment]
@samdark  sure but what docs should be updated and with what? are we talking about the DocBock of `yii\db\Connection` ?
```php
    /**
     * @var bool whether to enable profiling of database queries. Defaults to true.
     * You may want to disable this option in a production environment to gain performance
     * if you do not need the information being logged.
     * @since 2.0.12
     * @see enableLogging
     */
```

**Comment by samdark** [comment]
Something like

> whether to enable profiling of opening database connection and database queries

File: framework/db/Connection.php [hunk 1]
```diff
@@ -577,12 +577,21 @@ public function open()
         $token = 'Opening DB connection: ' . $this->dsn;
         try {
             Yii::info($token, __METHOD__);
-            Yii::beginProfile($token, __METHOD__);
+            if ($this->enableProfiling) {
+                Yii::beginProfile($token, __METHOD__);
+            }
+
             $this->pdo = $this->createPdoInstance();
             $this->initConnection();
```
Comment (cebe) [inline framework/db/Connection.php]: there is a small chance that someone is using `initConnection` to configure the db connection, i.e. setting `enableProfiling` to another value. To make the code more robust I would get the value `$enableProfiling = $this->enableProfiling` and use the local variable afterwards to keep the nesting of profiling calls.

**Review by cebe — CHANGES_REQUESTED** [review]

File: framework/db/Connection.php (hunk 1 above)
Comment (njasm) [reply framework/db/Connection.php]: sure.. I've missed that. 
done! 
/cc @cebe @samdark @rob006

**Review by cebe — APPROVED** [review]

**Comment by samdark** [comment]
Merged. Thanks!

**Pull request merged**